File: ITODSocialGraphs.Server/Models/Graph.cs
namespace ITODSocialGraphs.Server.Models
{
    public class Graph
    {
        public List<Node> Nodes { get; set; } = new();
        public List<Edge> Edges { get; set; } = new();
    }

    public class Node
    {
        public string Id { get; set; }
        public string Name { get; set; }
    }

    public class Edge
    {
        public string Source { get; set; }
        public string Target { get; set; }
    }
}

File: itodsocialgraphs.client/src/types/graph.ts
export interface Node {
    id: string;
    name: string;
    x?: number;
    y?: number;
    vx?: number;
    vy?: number;
    fx?: number | null;
    fy?: number | null;
  }
  
  export interface Edge {
    source: string | Node;
    target: string | Node;
  }
  
  export interface Graph {
    nodes: Node[];
    edges: Edge[];
  }
  